src/api/source/TileClient.js

```diff
@@ -0,0 +1,80 @@
+import { validateTemplateURL } from '../url';
+import DataframeCache from './DataframeCache';
+import { rTiles } from '../../client/rsys';
+
+
+export default class TileClient {
+    constructor(templateURLs) {
+        if (!Array.isArray(templateURLs)) {
+            templateURLs = [templateURLs];
+        }
+        templateURLs.map(validateTemplateURL);
+        this._templateURLs = templateURLs;
+        this._requestGroupID = 0;
+        this._oldDataframes = [];
+        this._cache = new DataframeCache();
+    }
+
+    bindLayer(addDataframe, dataLoadedCallback) {
+        this._addDataframe = addDataframe;
+        this._dataLoadedCallback = dataLoadedCallback;
+    }
+
+    requestData(viewport, responseToDataframeTransformer) {
+        const tiles = rTiles(viewport);
+        this._getTiles(tiles, responseToDataframeTransformer);
+    }
+
+    free() {
+        this._cache = new DataframeCache();
+        this._oldDataframes = [];
+    }
+
+    _getTiles(tiles, responseToDataframeTransformer) {
+        this._requestGroupID++;
+        let completedTiles = [];
+        let needToComplete = tiles.length;
+        const requestGroupID = this._requestGroupID;
+        tiles.forEach(t => {
```

Comment:
tile

Reply:
I think it's pretty clear that it refers to a tile, it is only used in the next line

Reply:
```js
tiles.forEach(({x, y, z}) => {
```

Reply:
I agree with @IagoLast that we should use descriptive names even if it is clear for us.